src/backend/distributed/transaction/multi_shard_transaction.c

```diff
@@ -167,12 +170,14 @@ BeginTransactionOnShardPlacements(uint64 shardId, char *userName)
 
 		MemoryContextSwitchTo(oldContext);
 
-		/* now that connection is tracked, issue BEGIN */
-		result = PQexec(connection, "BEGIN");
-		if (PQresultStatus(result) != PGRES_COMMAND_OK)
-		{
-			ReraiseRemoteError(connection, result);
-		}
+		/* issue BEGIN */
+		RemoteTransactionBegin(connection);
+
+		/*
+		 * Every individual failure should cause entire distributed
+		 * transaction to fail.
+		 */
+		MarkRemoteTransactionCritical(connection);
```

Comment:
Shouldn't this be set before the begin? 

Also, what if we call this automatically using something like a `TRANSACTION_CRITICAL` flag?